flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/FsStateChangelogWriter.java

```diff
@@ -205,7 +205,7 @@ public SequenceNumber lastAppendedSequenceNumber() {
             SequenceNumberRange range = SequenceNumberRange.generic(from, activeSequenceNumber);
             if (range.size() == readyToReturn.size()) {
                 checkState(toUpload.isEmpty());
-                return completedFuture(buildHandle(keyGroupRange, readyToReturn));
+                return completedFuture(buildHandle(keyGroupRange, readyToReturn, 0L));
```

Comment:
Similar problem: in case if **all** state was pre-uploaded, the reported size will be zero.

Reply:
As discussed above, this is by design.